### PR TITLE
Manual ip addresses

### PR DESCRIPTION
**Pull request by arodier**
Removed all the dynamic ip address detection, factorised ip address type detection, use ipv6 filter from Ansible. Cleaner code.

### CODE REVIEW

File: install/playbooks/main.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@
 # Prepare the system, get its external IP address,
 # and set up remote access
 - import_playbook: system-prepare.yml
-- import_playbook: external-ip.yml
+- import_playbook: external-ip-type.yml
```
Comment (fredericmoulins) [inline install/playbooks/main.yml]: If the playbook is removed, this is not needed anymore. Each playbook will have a dependency on the common `external-ip-type` role.

File: common/roles/external-ip-type/tasks/main.yml [hunk 2]
```diff
@@ -8,11 +8,6 @@
...
 
 - name: Get the external IP address automatically
-  when: network.auto_detect_ip == "auto"
-  include_tasks: auto-ip.yml
-
-- name: Get the external IP address automatically
-  when: network.auto_detect_ip != "auto"
   include_tasks: manual-ip.yml
 
 # Detect the main IP address type (IPv4 or IPv6)
```
[the remark this answers is not in the document]
Comment (arodier) [reply common/roles/external-ip-type/tasks/main.yml]: Yes, this is what I wanted. Thanks for the review !

File: common/roles/external-ip-type/tasks/manual-ip.yml [hunk 3]
```diff
@@ -7,8 +7,20 @@
   set_fact:
     external_ip: '{{ network.external_ip }}'
 
+- name: Set external IP address type (A or AAAA)
+  tags: facts
+  set_fact:
+    external_ip_type: '{{ external_ip | ipv6() | ternary("AAAA", "A") }}'
```
[the remark this answers is not in the document]
Comment (arodier) [reply common/roles/external-ip-type/tasks/manual-ip.yml]: Thanks for this. I am not planning to support other than Debian and Ubuntu. Even Ubuntu, I don't see the advantages.
Thanks as well for the reverse DNS entry, very useful.

File: common/roles/external-ip-type/tasks/manual-ip.yml [hunk 4]
```diff
@@ -7,8 +7,20 @@
   set_fact:
     external_ip: '{{ network.external_ip }}'
```
[the remark this answers is not in the document]
Comment (arodier) [reply common/roles/external-ip-type/tasks/manual-ip.yml]: Yes, that's a good idea. Will do

File: common/roles/external-ip-type/tasks/manual-ip.yml [hunk 5]
```diff
@@ -7,8 +7,20 @@
...
 
+- name: Set external IP address type (A or AAAA)
+  tags: facts
+  set_fact:
+    external_ip_type: '{{ external_ip | ipv6() | ternary("AAAA", "A") }}'
+
 # Backup IP address
-- name: Get the backup IP address specified manually
+- name: Get the backup IP address
+  when: network.backup_ip != None
```
[the remark this answers is not in the document]
Comment (arodier) [reply common/roles/external-ip-type/tasks/manual-ip.yml]: Idem,

File: common/roles/external-ip-type/tasks/manual-ip.yml [hunk 6]
```diff
@@ -7,8 +7,20 @@
...
   tags: facts
   set_fact:
     backup_ip: '{{ network.backup_ip }}'
+
+- name: Set external IP address type (A or AAAA)
+  tags: facts
+  when: network.backup_ip != None
```
[the remark this answers is not in the document]
Comment (arodier) [reply common/roles/external-ip-type/tasks/manual-ip.yml]: Will do too

File: install/playbooks/external-ip-type.yml [hunk 7]
```diff
@@ -6,4 +6,4 @@
     - '{{ playbook_dir }}/../../config/system.yml'
```
[the remark this answers is not in the document]
Comment (arodier) [reply install/playbooks/external-ip-type.yml]: Well spotted again! Great.

File: install/playbooks/roles/external-ip/defaults/main.yml [hunk 8]
```diff
@@ -1,5 +0,0 @@
----
-
-ip_check_packages:
-  - geoip-bin
-  - geoip-database
```
[the remark this answers is not in the document]
Comment (arodier) [reply install/playbooks/roles/external-ip/defaults/main.yml]: Yes, absolutely. You are right.

File: install/playbooks/main.yml (hunk 1 above)
Comment (arodier) [reply install/playbooks/main.yml]: Yes, we can use the meta on this.

**Comment by arodier** [comment]
A lot to do, that's great.

**Comment by arodier** [comment]
Perhaps I did not push the last version, some modifications were already made. Still, it is interesting to implement the rest.

File: common/roles/external-ip-type/tasks/manual-ip.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (arodier) [reply common/roles/external-ip-type/tasks/manual-ip.yml]: For the sanity check, I will create a role dedicated to the basics, before running. Maybe. Perhaps another pull request with multiple sanity checks.

**Comment by arodier** [comment]
OK, Once I will have finish the testing, I will come back to you on this one.

**Comment by arodier** [comment]
Work in progress, I will tell when it will be ready to merge.

**Comment by arodier** [comment]
Hello @fredericmoulins . I think it is now ready to merge. Maybe not perfect, but better than before, and I do not want to wait too long, as I have other plans coming.
Thanks.

File: install/playbooks/roles/system-prepare/defaults/main.yml [hunk 9]
```diff
@@ -14,11 +14,12 @@ system_required:
   - nginx
   - acl
   - certbot
+  - python-netaddr
```
Comment (fredericmoulins) [inline install/playbooks/roles/system-prepare/defaults/main.yml]: I don't think this is needed for the target server.
I wasn't specific enough in my comment: the dependency I was talking about is on the host running the ansible playbooks.
As it concerns only the ansible host, I thought an addition to the docs would be sufficient. Currently, in `docs/prerequisites.md` there is an example that will still be valid (as ansible depends on python3-netaddr on debian and ubuntu). But for other platforms, the installation of the python `netaddr` module might be required.

File: install/playbooks/roles/system-cleanup/defaults/main.yml [hunk 10]
```diff
@@ -9,3 +9,4 @@
 cleanup_packages:
   - bash-completion
   - resolvconf
+  - python-netaddr
```
Comment (fredericmoulins) [inline install/playbooks/roles/system-cleanup/defaults/main.yml]: Not needed IMO. Please see my other comment.

File: common/roles/external-ip-type/tasks/manual-ip.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (fredericmoulins) [reply common/roles/external-ip-type/tasks/manual-ip.yml]: A sanity check at the beginning of any run would work for me.

File: common/roles/external-ip-type/tasks/manual-ip.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (fredericmoulins) [reply common/roles/external-ip-type/tasks/manual-ip.yml]: Might be done later. Not important.

**Comment by arodier** [comment]
Thanks, I will do!

**Pull request merged**